### DISCUSSION TITLE
Import Vue component in typescript file

### DISCUSSION THREAD

**Discussion by andredewaard** [zed-industries/zed#6970] [Feature Requests]
Don't want to create an issue right away since maybe this is just a setting somewhere. 
Importing a Vue component inside another Vue component is working but doing so inside an typescript/javascript file doesn't.
This does work in VSCode so i guess it has something to do with the typescript language server not recognising Vue extension.

<img width="1005" alt="Screenshot 2024-01-29 at 09 21 11" src="https://github.com/zed-industries/zed/assets/7560637/2d661c02-8aa7-4dea-9caa-4db71e0b7340">
<img width="1471" alt="Screenshot 2024-01-29 at 09 22 19" src="https://github.com/zed-industries/zed/assets/7560637/6068d042-c329-4c21-a7df-6ea41ec30375">

**Comment by JDIZM** [comment]
I was going to raise this myself, it looks like zed needs a plugin to support TS/Vue.

Taken from one of the readme's from a Vue project setup:

> TypeScript cannot handle type information for `.vue` imports by default, so we replace the `tsc` CLI with `vue-tsc` for type checking. In editors, we need [TypeScript Vue Plugin (Volar)](https://marketplace.visualstudio.com/items?itemName=Vue.vscode-typescript-vue-plugin) to make the TypeScript language service aware of `.vue` types.

Vue has some docs on it with a recommended IDE setup. https://vuejs.org/guide/typescript/overview

It would be good if Zed supported this and we can get it on the recommended list.

**Reply by andredewaard** [reply]
This is definitely the problem, Zed is unusable for me in combination with Vue as long this isn't fixed.

**Comment by Moshyfawn** [comment]
There's an issue raised for this in #8336.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
